### PR TITLE
Code review

### PR DESCRIPTION
**Pull request by Codenivore**

### CODE REVIEW

**Comment by Codenivore** [comment]
Overall this is a great piece of work Heli, I hope you are as proud of yourself as I am of you! I have added a few pointers in there for potential improvements and things to think about. Throughout though everything is nicely laid out and is a pleasure to read as well as adding relevant and appropriate comments where relevant too. Great work mate, well done you! :D Learned so much over the last 6 weeks!!!!!!!!!

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,131 @@
+node_modules
```
[the remark this answers is not in the document]
Comment (heliDevine) [reply .gitignore:1]: This .gitignore file got generated with command from the track material. I can find it.. same thing happend with the Music-library and Stu said similar thing in the review. So no, i don't know what all of this meant!

File: README.md [hunk 2]
```diff
@@ -1 +1,66 @@
...
+
+The purpose of this project was to deepend learned knowledge and build an API app with sequelize to manage a book library using CRUD operations.
+
+**Set up**
+
+To be able to run this application following environment needs to be installed:
+
+-clone this repo and set up local .env settings for:
+
+DB_PASSWORD=password
```
[the remark this answers is not in the document]
Comment (heliDevine) [reply README.md:13]: Ok, great and good to know that about the DB port details etc.

File: src/app.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+const express = require('express');
+
+const readerRouter = require('./routes/reader');
+const bookRouter = require('./routes/book');
+
+const app = express();
+app.use(express.json());
+
+app.get('/', (req, res) => {
```
[the remark this answers is not in the document]
Comment (heliDevine) [reply src/app.js:9]: Ok, i see! all noted.

File: src/controllers/book.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
+const { Book } = require("../models");
+
```
[the remark this answers is not in the document]
Comment (heliDevine) [reply src/controllers/book.js:2]: This was obviously one of the hardest parts of the project so when I got them working I was too scared to change anything. The findAll looked so pretty with .then syntax and I learned it from the other tutorial so I wanted to leave that in.

File: src/controllers/book.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+		res.status(201).json(newBook);
+	} catch (err) {
+		if (err.name === "SequelizeValidationError") {
+			res.status(400).json({ errors: err.errors });
+		} else {
+			res.status(500).json(err);
+		}
+	}
+};
+
```
[the remark this answers is not in the document]
Comment (heliDevine) [reply src/controllers/book.js:16]: I'm not exactly sure how to do that? I think I need to look back to the music-library.

File: src/controllers/reader.js [hunk 6]
```diff
@@ -0,0 +1,81 @@
+const { Reader } = require('../models');
+
+/// CREATE new reader
+exports.create = async (req, res) => {
+	try {
+		const newReader = await Reader.create(req.body);
+
+		res.status(201).json(`reader ${newReader.name} created`);
```
[the remark this answers is not in the document]
Comment (heliDevine) [reply src/controllers/reader.js:8]: That's what I thought that the test was the limiting factor as It worked in Postman. I wasn't quite sure how to change test.

**Pull request merged**